### PR TITLE
Pagination for Fragmentarium Search Results

### PR DESCRIPTION
**Pull request by yCobanoglu**

### CODE REVIEW

**Comment by ejimsan** [comment]
[Add pager to results](https://trello.com/c/ljzBhyVK/604-add-pager-to-results)

File: src/fragmentarium/ui/search/PaginationItems.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  ) {
+    paginationItems = (
+      <Pagination>{[first, ellipsis, ...itemComponents, last]}</Pagination>
+    )
+  }
+  if (items.length < 2 * NEIGHBOURING_PAGINATION_ITEMS) {
+    paginationItems = <Pagination>{[...itemComponents]}</Pagination>
+  }
+  return paginationItems
+}
```
[the remark this answers is not in the document]
Comment (fsimonjetz) [reply src/fragmentarium/ui/search/PaginationItems.tsx:138]: Logically, the second (`if (items[0].index === 1)`) and last conditions could also be `elif`, I guess that would be (marginally) more efficient(?)

[the remark this answers is not in the document]
Comment (yCobanoglu) [reply src/fragmentarium/ui/search/PaginationItems.tsx:138]: I will incorporate it.

File: src/fragmentarium/ui/search/FragmentariumSearchResult.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        const uncertain = genreItem.uncertain ? '(?)' : ''
+        return (
+          <ul key={genreItem.toString}>
+            <small>{`${genreItem.category.join(' ➝ ')} ${uncertain}`}</small>
+          </ul>
+        )
+      })}
+    </ul>
+  )
+}
```
Comment (khoidt) [inline src/fragmentarium/ui/search/FragmentariumSearchResult.tsx:66]: The layout of the search page breaks: 
image

**Review by khoidt — CHANGES_REQUESTED** [review]
Please fix the layout issue and satisfy CodeClimate (the "similar blocks" can be ignored).

File: src/fragmentarium/ui/search/FragmentariumSearchResult.tsx (hunk 2 above)
Comment (yCobanoglu) [reply src/fragmentarium/ui/search/FragmentariumSearchResult.tsx:66]: fixed it

File: src/fragmentarium/ui/search/FragmentariumSearchResults.test.tsx [hunk 3]
```diff
@@ -104,14 +111,52 @@ describe('search fragmentarium only transliteration', () => {
...
       expect(screen.getAllByText(`(${fragment.script})`)).not.toEqual([])
     }
   })
-
   it('Displays matching lines', () => {
     expect(screen.getByText('kur')).toBeVisible()
   })
 })
+
+let CUT_OFF: number
```
Comment (fsimonjetz) [inline src/fragmentarium/ui/search/FragmentariumSearchResults.test.tsx:124]: `CUT_OFF` is basically the number of items per page, right?

**Review by fsimonjetz — APPROVED** [review]

File: src/fragmentarium/ui/search/FragmentariumSearchResults.test.tsx (hunk 3 above)
Comment (yCobanoglu) [reply src/fragmentarium/ui/search/FragmentariumSearchResults.test.tsx:124]: yes

**Pull request merged**